docs/blocks/block-migrations.mdx

```diff
@@ -0,0 +1,207 @@
+        return {
+            ...other,
+            eyebrow: {
+                draftContent: {
+                    blocks: [
+                        {
+                            key: "12345",
+                            // The "old" eyebrow string
+                            // highlight-next-line
+                            text: eyebrow,
+                            type: "unstyled",
+                            depth: 0,
+                            inlineStyleRanges: [],
+                            entityRanges: [],
+                            data: {},
+                        },
+                    ],
+                    entityMap: {},
+                },
+            },
+        };
+    }
+}
+```
+
+Finally, we add the migration to the block using the `migrate` option in `createBlock`.
+
+```ts
+import { createBlock, typesafeMigrationPipe } from "@comet/blocks-api";
+
+...
+
+export const HeadlineBlock = createBlock(HeadlineBlockData, HeadlineBlockInput, {
+    name: "Headline",
+    migrate: {
+        version: 1,
+        // highlight-next-line
+        migrations: typesafeMigrationPipe([ChangeEyebrowMigration]),
+    },
+});
+```
+
+The next time the block is loaded from the database, the migration will be performed, migrating the block to the new structure.
+
+### Complete migration class
+
+```ts title="1-change-eyebrow.migration.ts"
+import { BlockDataInterface, BlockMigration, BlockMigrationInterface } from "@comet/blocks-api";
+
+interface From {
+    headline: any;
+    eyebrow: string;
+    level: any;
+}
+
+interface To {
+    headline: any;
+    eyebrow: BlockDataInterface;
+    level: any;
+}
+
+export class ChangeEyebrowMigration extends BlockMigration<(from: From) => To> implements BlockMigrationInterface {
+    public readonly toVersion = 1;
+
+    protected migrate({ eyebrow, ...other }: From): To {
+        return {
+            ...other,
+            eyebrow: {
+                draftContent: {
+                    blocks: [
+                        {
+                            key: "12345",
+                            text: eyebrow,
+                            type: "unstyled",
+                            depth: 0,
+                            inlineStyleRanges: [],
+                            entityRanges: [],
+                            data: {},
+                        },
+                    ],
+                    entityMap: {},
+                },
+            },
+        };
+    }
+}
+```
+
+## FAQ
+
+### When do I need a migration?
+
+You will need a migration when
+
+-   **adding** a _required_ field to the block
+-   **changing** an existing field of the block
+
+You don't necessarily need a migration when
+
+-   **adding** an _optional_ field to the block
+-   **removing** an existing field from the block
+-   **adding** an additionally supported block to a blocks block or an one of block
+-   **removing** a supported block from a blocks block or an one of block
```

Comment:
Maybe use PascalCase for `BlocksBlock` and `OneOfBlock`?

Otherwise it seems hard to read and it may be confusing to new Comet users

Reply:
You're right! Changed in 6fd27b1a584eccbbd78341ccfb0a1a38cb881b3c